overlord/ifacestate/handlers.go

```diff
@@ -96,9 +96,15 @@ func (m *InterfaceManager) doSetupProfiles(task *state.Task, _ *tomb.Tomb) error
 		return err
 	}
 	if len(affectedSnaps) == 0 {
-		affectedSnaps = append(affectedSnaps, snapInfo)
+		affectedSnaps = append(affectedSnaps, snapName)
 	}
-	for _, snapInfo := range affectedSnaps {
+	for _, snapName := range affectedSnaps {
+
+		snapInfo, err := snapstate.Current(task.State(), snapName)
+		if err != nil {
+			return err
+		}
+		snap.AddImplicitSlots(snapInfo)
```

Comment:
this looks strange? if it's always needed shouldn't something in snap do this instead when we make Infos? otherwise shouldn't there be an helper in ifacestate that gets the slots from an info and fills in the implicit ones then?